### ISSUE TITLE
[Notification] does not unmount fully

### ISSUE THREAD

**Issue by ah-arch** [telus/tds-core#1295]
## Description

- When using the `dismissible` prop with the `Notification` component, changing the state after dismissing the notification would cause the page to crash with the console error:
```TypeError: Cannot read property 'offsetHeight' of null```

## Reproduction Steps

1. Test example with a `Button` that changes state and a dismissible `Notification`:
```
class Example extends React.Component {
  render() {
    return (
      <>
        <Button onClick={() => this.setState({ any: 'test' })}>Click</Button>
        <Notification copy="en" dismissible>Notification</Notification>
      </>
    )
  }
};
<Example />
```
2. Click on the button which updates the state.
3. Dismiss the notification.
4. Observe the error:
<img width="637" alt="TypeError: Cannot read property 'offsetHeight' of null" src="https://user-images.githubusercontent.com/4450690/67425311-c11fb380-f5a5-11e9-839d-e3361bb57be6.png">

## Workaround details

Not sure if there is a workaround.

## Recommendation

Quick workaround by checking for `this.contentWrapper` before checking `this.contentWrapper.offsetHeight`:
https://github.com/telus/tds-core/blob/06dddb2ac11b30825de6397aa52f5e97fb65f8c1/packages/Notification/Notification.jsx#L95

## Meta
- TDS component version: `@tds/core-notification@3.0.3`
- Willing to develop solution: Yes
- High impact: Yes

**Comment by theetrain** [comment]
Note, the issue may be with Notification not unmounting properly, causing [`componentDidUpdate`](https://github.com/telus/tds-core/blob/f0ea3e2b7319b5379317de2ba72f3cd6c682179e/packages/Notification/Notification.jsx#L88-L92) to fire even after it has dismissed.

**Comment by ah-arch** [comment]
> Note, the issue may be with Notification not unmounting properly, causing [`componentDidUpdate`](https://github.com/telus/tds-core/blob/f0ea3e2b7319b5379317de2ba72f3cd6c682179e/packages/Notification/Notification.jsx#L88-L92) to fire even after it has dismissed.

Perhaps the `componentDidUpdate` and `componentDidMount` checks here:
https://github.com/telus/tds-core/blob/06dddb2ac11b30825de6397aa52f5e97fb65f8c1/packages/Notification/Notification.jsx#L89-L91

Could be updated to:
```
this.props.dismissible && !this.state.dismissed
```

**Comment by jraff** [comment]
@ah-arch Fixed with [@tds/core-notification@3.0.4](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-notification%403.0.4)

**Comment by ah-arch** [comment]
Well the issue is remains as it does not unmount fully.

If it is placed in `<Box between={3} />` for example and then dismissed, it would still retain its bottom margin due to the 2 visible wrappers.

**Comment by jraff** [comment]
Followup issue created #1298

**Comment by marcod1419** [comment]
This has been addressed in Notification 3.0.7: https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-notification%403.0.7
